pkg/pod/v1alpha1/pod.go

```diff
@@ -0,0 +1,111 @@
+package v1alpha1
+
+import v1 "k8s.io/api/core/v1"
+
+// pod holds the api's pod objects
+type pod struct {
+	object *v1.Pod
+}
+
+// PodList holds the list of pod instances
+type PodList struct {
+	Items []*pod
+}
+
+// listBuilder enables building an instance of
+// podlist
+type listBuilder struct {
+	list    *PodList
+	filters predicateList
+}
+
+// WithAPIList builds the list of pod
+// instances based on the provided
+// pod list api instance
+func (b *listBuilder) WithAPIList(pods *v1.PodList) *listBuilder {
+	if pods == nil {
+		return b
+	}
+	for _, p := range pods.Items {
+		b.list.Items = append(b.list.Items, &pod{&p})
+	}
+	return b
+}
+
+// WithAPIList builds the list of pod
+// instances based on the provided
+// pod api instances
+func (b *listBuilder) WithAPIObject(pods ...v1.Pod) *listBuilder {
+	if len(pods) == 0 {
+		return b
+	}
+	for _, p := range pods {
+		b.list.Items = append(b.list.Items, &pod{&p})
+	}
+	return b
+}
+
+// List returns the list of pod
+// instances that was built by this
+// builder
+func (b *listBuilder) List() *PodList {
+	if b.filters == nil && len(b.filters) == 0 {
+		return b.list
+	}
+	filtered := ListBuilder().List()
+	for _, pod := range b.list.Items {
+		if b.filters.all(pod) {
+			filtered.Items = append(filtered.Items, pod)
+		}
+	}
+	return b.list
+}
+
+// ToAPIList converts podList to API podList
+func (p *PodList) ToAPIList() *v1.PodList {
```

Comment:
Along with this function we should have 
listBuilder expose APIList method.
e.g.
```go
func (l *listBuilder) APIList() *v1.PodList{
   // make use of l.List() & other logic
}
```